packages/react/src/provider/use-context-mutator.ts

```diff
@@ -0,0 +1,26 @@
+import { useContext } from 'react';
+import { OpenFeature, EvaluationContext } from '@openfeature/web-sdk';
+import { Context } from './context';
+
+/**
+ *
+ * A hook for accessing context mutating functions.
+ *
+ */
+export function useContextMutator() {
+  const { domain } = useContext(Context) || {};
+
+  async function mutateContext(updatedContext: EvaluationContext): Promise<void> {
+    if (!domain) {
+      // Set the global context
+      OpenFeature.setContext(updatedContext);
+      return;
+    }
+
+    OpenFeature.setContext(domain, updatedContext);
+  }
+
+  return {
+    mutateContext,
+  };
+}
```

Comment:
```suggestion
export function useContextMutator() {
    const { domain } = useContext(Context) || {};
    const previousContext = useRef(null);

    const mutateContext = useCallback(async (updatedContext: EvaluationContext) => {
        if (previousContext.current !== updatedContext) {
            if (!domain) {
                OpenFeature.setContext(updatedContext);
            } else {
                OpenFeature.setContext(domain, updatedContext);
            }
            previousContext.current = updatedContext;
        }
    }, [domain]);

    return {
        mutateContext,
    };
}
```

I would like to consider two things here. 

First is to memoize the `mutateContext` function here to avoid re-rendering of dependent components if they pass is downwards. This can just prevent some unnecessary re-renders.

Second and more important is that calling `setContext` can be quite expensive. The reason is outlined [here](https://openfeature.dev/blog/reconciling-with-state/#context-reconciliation) and specified [here](https://openfeature.dev/specification/sections/providers#26-provider-context-reconciliation), but so summarize, in the `static context paradigm` which the Web SDK and so React SDK use, setting the context can lead to a provider refreshing it's cached flags which most times means calling APIs.  
To prevent this being called unnecessarily, we should at least check if the new context is the same we had before. 
In my suggestion I am only checking for referential equality, maybe can deep equal this (cc @beeme1mr @toddbaert ), but I could imagine referential equality being good enough for this problem.

Reply:
@lukas-reining I had memoized callback ready for my next commit :) As for checking the object, I think it makes sense and can be added, but probably have to do a deep comparison as we can't assume that the developer will be using the same object reference when they call `setContext`. I found in react-land people like to copy objects everywhere.

Reply:
I feel like that logic would be better in the web sdk and could be handled in a separate pr.

Reply:
For the deep equality I am with you @beeme1mr. Checking for referential equality should be done directly as even successive renders would lead to reconciliation otherwise.

Reply:
Got it, deep equal in the core lib, and a simple reference check in react to prevent some rerenders.